src/System.Runtime.Extensions/src/System/AppDomain.cs

```diff
@@ -26,5 +27,147 @@ public partial class AppDomain
+        public object GetData(string name) => AppContext.GetData(name);
+        // Should this move to AppContext
+        public void SetData(string name, object data) => AppDomainAugments.SetData(name, data);
+        public System.Nullable<bool> IsCompatibilitySwitchSet(string value) => AppDomainAugments.IsCompatibilitySwitchSet(value);
+        public bool IsDefaultAppDomain()  => true;
+        public bool IsFinalizingForUnload() => false;
+        public void SetCachePath(string path) { throw new PlatformNotSupportedException(); }
+        public void SetPrincipalPolicy(System.Security.Principal.PrincipalPolicy policy) { throw new PlatformNotSupportedException(); }
+        public void SetShadowCopyFiles() { throw new PlatformNotSupportedException(); }
+        public void SetShadowCopyPath(string path) { throw new PlatformNotSupportedException(); }
+        public void SetThreadPrincipal(System.Security.Principal.IPrincipal principal) { throw new PlatformNotSupportedException(); }
+        public override string ToString() => AppDomainAugments.ToString();
+        public static void Unload(System.AppDomain domain) { throw new PlatformNotSupportedException(); }
+
+        public System.Reflection.Assembly Load(byte[] rawAssembly) => AppDomainAugments.Load(rawAssembly, null);
+        public System.Reflection.Assembly Load(byte[] rawAssembly, byte[] rawSymbolStore) => AppDomainAugments.Load(rawAssembly, rawSymbolStore);
+        public System.Reflection.Assembly Load(System.Reflection.AssemblyName assemblyRef) => AppDomainAugments.Load(assemblyRef);
+        public System.Reflection.Assembly Load(string assemblyString) => AppDomainAugments.Load(assemblyString);
+
+
+        public event System.AssemblyLoadEventHandler AssemblyLoad
+        {  
+            add  
+            {  
+                AppDomainAugments.AssemblyLoad += value;  
+            }  
+  
+            remove  
+            {  
+                AppDomainAugments.AssemblyLoad -= value;  
+            }  
+        }  
+
+        public event System.ResolveEventHandler TypeResolve
+        {  
+            add  
+            {  
+                AppDomainAugments.TypeResolve += value;  
+            }  
+  
+            remove  
+            {  
+                AppDomainAugments.TypeResolve -= value;  
+            }  
+        } 
+
+        public event System.ResolveEventHandler ResourceResolve
+        {  
+            add  
+            {  
+                AppDomainAugments.ResourceResolve += value;  
+            }  
+  
+            remove  
+            {  
+                AppDomainAugments.ResourceResolve -= value;  
+            }  
+        } 
+        
+        public static bool MonitoringIsEnabled
+        {  
+            get  
+            {  
+                return AppDomainAugments.MonitoringIsEnabled;
+            }  
+  
+            set
+            {  
+                AppDomainAugments.MonitoringIsEnabled = true;
```

Comment:
In accordance with the desktop framework, it should even go further: throw an ArgumentException if "false" would be attempted to set (you can only enable monitoring, never disable it).

http://referencesource.microsoft.com/#mscorlib/system/appdomain.cs,4752